### PR TITLE
Add documentation on exit status codes to man pages

### PR DESCRIPTION
**Pull request by rigtorp**
This is step one to start defining useful exit codes for all the commands.

What is the purpose of this change? What does it change?
--------------------------------------------------------

Add documentation on restic exit status codes to all man pages.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Yes, it's related to #2364 but doesn't resolve the issue.

Checklist
---------

- [X] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [ ] I have added tests for all changes in this PR
- [X] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [ ] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

**Comment by rawtaz** [comment]
@rigtorp Thanks for your contribution, a small but good one!

I have a couple of questions:

1. Restic can return other exit codes than 0 and 1, so could you reword the text to be something like "Exit status is 0 if the command was successful, and non-zero if there was any error." instead?

2. Is there any particular reason why the "EXIT STATUS" is quoted, when SEE ALSO is not? Should the quotes be removed, or should the other one have them added?

Since this branch isn't merged, you can force-push amended commits to it if you want.

**Comment by rigtorp** [comment]
>     1. Restic can return other exit codes than 0 and 1, so could you reword the text to be something like "Exit status is 0 if the command was successful, and non-zero if there was any error." instead?

No Restic can only return 0 or 1 currently. It would great to improve that in the future, for example to return a specific error code when backup completed with warnings vs backup failed. See #2364 

>
>     2. Is there any particular reason why the "EXIT STATUS" is quoted, when SEE ALSO is not? Should the quotes be removed, or should the other one have them added?

I've never thought about it. I've always quoted the section headers. I will remove the quotes and verify that the output still looks correct. Better stick to the existing style.

**Comment by rigtorp** [comment]
Restic exit status:
https://github.com/restic/restic/blob/master/cmd/restic/main.go#L107

For reference:
- http://www.tldp.org/LDP/abs/html/exitcodes.html
- https://github.com/openbsd/src/blob/master/include/sysexits.h
- https://www.gnu.org/software/libc/manual/html_node/Exit-Status.html

**Comment by rigtorp** [comment]
Actually I found this code where Restic can exit with 130:
https://github.com/restic/restic/blob/master/cmd/restic/cleanup.go#L69

**Comment by rawtaz** [comment]
Thanks for elaborating :) I'm certainly not always right, very open to being proven wrong.

Yeah, that one 130, and there's also a 2, 3 and 5 if you enable debug.go I think.

If you don't mind I'd like to change the text, because that will also make it more future-proof, for whenever someone forgets to change the EXIT STATUS text properly down the line.

**Comment by rigtorp** [comment]
I'm going to change it to this:
```
Exit status is 0 if the command was successful, and non-zero if there was any error.

0    Successful program execution.
1    Generic error.
130  Program execution was interrupted with SIGINT.
```

Later we can use `errors.Is()` to query for IO, DNS and other errors and return according to `/usr/include/sysexits.h`

**Comment by rawtaz** [comment]
Fair enough - after all, having the various possible exit codes documented is what this is about.

**Comment by fd0** [comment]
Thanks for your contribution! Unfortunately the man pages are automatically generated from the descriptions in the source code, usually before a release (so they are up to date). One example for a man page text is here:

https://github.com/restic/restic/blob/26d1f9f4ba6c106a90e2af9a96df50d8898a9bb5/cmd/restic/cmd_check.go#L23-L27

You can generate the manpages by building restic and then running `mkdir manpages && ./restic generate --man manpages`. They can be viewed using `man` like this: `man ./manpages/restic-check.1`. Inserting a new section works like this (I think it's markdown):

```go
var cmdCheck = &cobra.Command{
	Use:   "check [flags]",
	Short: "Check the repository for errors",
	Long: `
The "check" command tests the repository for errors and reports any errors it
finds. It can also be used to read all data and therefore simulate a restore.

By default, the "check" command will always load all data directly from the
repository and not use a local cache.

EXIT STATUS
===========

Exit status is 0 if command was successful, and 1 if there was any error.
`,
	DisableAutoGenTag: true,
```

**Comment by rigtorp** [comment]
@fd0 Eventually the exit status will diverge between the commands. In particular I want to make the backup command return different status depending on if backup completed without warnings or it failed. Currently it returns 0 when backup completed with warnings.

**Comment by rawtaz** [comment]
@rigtorp Maybe it would be a good idea to make that (separate exit codes) a new issue for tracking, and fix this PR to reflect the current state with @fd0's comments above in mind, so we can merge this.

**Comment by rigtorp** [comment]
@rawtaz Yes that's what I had in mind.

**Comment by rigtorp** [comment]
@fd0 I have addressed all concerns. Specifically I added a note about current behavior of backup command.

File: changelog/unreleased/pull-2461
Comment (rawtaz) [inline changelog/unreleased/pull-2461]: Hm, does Calens (the changeling generator) work without the middle paragraph?

File: cmd/restic/cmd_restore.go
Comment (rawtaz) [inline cmd/restic/cmd_restore.go]: This change is unrelated and should be removed. I can try to do that when I have some time if you are fed up with making corrections to this PR by now :)

Comment (rigtorp) [reply cmd/restic/cmd_restore.go]: Fixed it, I think it was go fmt or something that changed it.

**Comment by rigtorp** [comment]
@rawtaz Friendly reminder!

**Comment by rigtorp** [comment]
@rawtaz @fd0 Can we merge this?

**Comment by rawtaz** [comment]
Looked at it earlier today, just waiting for a verification on something before we merge.

But we should remove the changelog file entirely. I'll do that if it's not done by whenever I merge it.

**Comment by rawtaz** [comment]
The following needs to be cleaned up, then this can be finally merged:

- All changes to files under `doc/man/` should be reverted/removed (man pages are generated upon release).
- All changes to files under `changelog/unreleased/` should be reverted/removed (not needed for doc changes).
- Squash commits into one.

**Review by rawtaz — APPROVED** [review]
I did the final polish. Thanks you very much Erik! Sorry it took so long.

**Pull request merged**
